src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/EnclaveDelegate.NetCoreApp.cs

```diff
@@ -82,5 +86,119 @@ internal byte[] GetSerializedAttestationParameters(SqlEnclaveAttestationParamete
+            {
+                switch (attestationProtocol)
+                {
+                    case SqlConnectionAttestationProtocol.AAS:
+                        AzureAttestationEnclaveProvider azureAttestationEnclaveProvider = new AzureAttestationEnclaveProvider();
+                        EnclaveProviders[attestationProtocol] = (SqlColumnEncryptionEnclaveProvider)azureAttestationEnclaveProvider;
+                        sqlColumnEncryptionEnclaveProvider = EnclaveProviders[attestationProtocol];
+                        break;
+
+                    case SqlConnectionAttestationProtocol.HGS:
+                        HostGuardianServiceEnclaveProvider hostGuardianServiceEnclaveProvider = new HostGuardianServiceEnclaveProvider();
+                        EnclaveProviders[attestationProtocol] = (SqlColumnEncryptionEnclaveProvider)hostGuardianServiceEnclaveProvider;
+                        sqlColumnEncryptionEnclaveProvider = EnclaveProviders[attestationProtocol];
+                        break;
+
+                    default:
+                        break;
+                }
+            }
+
+            if (sqlColumnEncryptionEnclaveProvider == null)
+            {
+                throw SQL.EnclaveProviderNotFound(enclaveType, ConvertAttestationProtocolToString(attestationProtocol));
+            }
+
+            return sqlColumnEncryptionEnclaveProvider;
+        }
+
+        private string ConvertAttestationProtocolToString(SqlConnectionAttestationProtocol attestationProtocol)
+        {
+            switch (attestationProtocol)
+            {
+                case SqlConnectionAttestationProtocol.AAS:
+                    return "AAS";
+
+                case SqlConnectionAttestationProtocol.HGS:
+                    return "HGS";
+
+                default:
+                    return "NotSpecified";
+            }
+        }
+
+        internal void GetEnclaveSession(SqlConnectionAttestationProtocol attestationProtocol, string enclaveType, string serverName, string enclaveAttestationUrl, out SqlEnclaveSession sqlEnclaveSession)
+        {
+            long counter;
+            GetEnclaveSession(attestationProtocol, enclaveType, serverName, enclaveAttestationUrl, out sqlEnclaveSession, out counter, throwIfNull: false);
+        }
+
+        // kz to move to netcoreapp
+        private void GetEnclaveSession(SqlConnectionAttestationProtocol attestationProtocol, string enclaveType, string serverName, string enclaveAttestationUrl, out SqlEnclaveSession sqlEnclaveSession, out long counter, bool throwIfNull)
+        {
+            SqlColumnEncryptionEnclaveProvider sqlColumnEncryptionEnclaveProvider = GetEnclaveProvider(attestationProtocol, enclaveType);
+            sqlColumnEncryptionEnclaveProvider.GetEnclaveSession(serverName, enclaveAttestationUrl, out sqlEnclaveSession, out counter);
+
+            if (throwIfNull)
+            {
+                if (sqlEnclaveSession == null)
```

Comment:
Can be simplified to `if (throwIfNull && !sqlEnclaveSession)`

Reply:
Yeah, combine two into one line will be better. But I don't think you can check null in this way in C#.

Reply:
there is a way to check strings in C#. Method string.IsNullOrEmpty(string) will check null strings.
To combine them we simply can write if (throwIfnull && !string.IsNullOrEmpty(sqlEnclaveSession))

Reply:
sqlEnclaveSession is not a string.